File: backend/backend.Domain/src/Entities/ReviewRate.cs
namespace backend.Domain.src.Entities
{
    public class ReviewRate : BaseEntity
    {
        public Guid UserId { get; set; }
        public Guid ProductId { get; set;}
        public string? Review { get; set; }
        public int? RatePoint { get; set; }

        public User User { get; set; }
        public Product Product { get; set; }
    }
}

File: frontend/src/types/ReviewRate.ts
import { Guid } from "guid-typescript";
import User from "./User";
import ProductDetail from "./ProductDetail";

export interface ReviewRateCreate
{
    userId: Guid
    productId: Guid
    review?: string
    ratePoint?: number
}

export interface ProductReviewRateRead
{
    user: User
    review?: string
    ratePoint?: number
}

export interface UserReviewRateRead
{
    product: ProductDetail
    review?: string
    ratePoint?: number
}

export interface ReviewRateRead
{
    productId: Guid
    review?: string
    ratePoint?: number
}
